README.md

```diff
@@ -6,3 +6,138 @@
 Cucumber helpers to make it easier to write living documentation in Ruby.
 
 See the documentation at https://www.relishapp.com/mojotech/dill/docs.
+
+#Overview
+Dill is designed to work seamlessly with Cucumber as a DOM interaction tool. Dill emphasizes relying on DOM attributes, rather than text, as attributes tend to be more stable.
+
+Dill might best be thought of as three layers:
+
+    ROLES, that perform => TASKS, which manipulate => ELEMENTS
+
+#A Dill Walkthrough
+For this walkthrough, we're going to build a single Cucumber step using Dill. If we use these concepts of Roles, Tasks and Elements, our cucumber step might end up looking something like this:
+
+```
+Then(^\the Admin should be able to create a new user$\) do
+  params = { name: 'Example', email: 'example@example.com' }
+
+  roles.admin.navigate_to_new_user
+  roles.admin.create_user(params)
+
+  expect(roles.admin).to see :new_user_message
+end
+```
+
+You can start to see in the expect statement that Dill is magical, too.
+
+##Roles
+A Dill role groups tasks and elements together, and is defined as a class.
+
+```ruby
+class Admin < Dill::Role
+  ...
+end
+```
+
+##Tasks
+A task is just a method on a role.
+
+```ruby
+class Admin < Dill::Role
+  def navigate_to_new_user
+    ...
+  end
+
+  def create_user(params)
+    ...
+  end
+
+  ...
+end
+```
+
+##Elements
+Elements are where the Dill amazingness begins, as well as where the Dill confusion heats up. Elements are dom abstractions that allow for precise testing. By using dom element attributes we can easily create the most basic Dill Element, a widget.
+
+```ruby
+class Admin < Dill::Role
+  widget :new_user_button, 'a.add-user'
+end
+```
+
+This widget uses css selectors to search the page for a link with the `add-user` class. That dom element is then wrapped into a Dill object, complete with a useful interface.
+
+```ruby
+  widget(:new_user_button).text # => 'Add a new user'
```

Comment:
I think there's a misunderstanding. There's nothing wrong in using parens here. In fact, [I think you should](http://blog.mojotech.com/a-fireside-chat-about-optional-parentheses/).